File: GoLang/quotes/svmService.go
package main

import (
	"context"
	"net/http"
	"time"

	pb "github.com/RYANCOAL9999/AI_Go_Proto_File"
	"github.com/gin-gonic/gin"
	"google.golang.org/grpc"
	"google.golang.org/grpc/credentials/insecure"
)

type svmGRPCClient struct {
	client pb.SVMServiceClient
}

func newSVMGRPCClient(addr string) (*svmGRPCClient, error) {
	conn, err := grpc.Dial(addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
	if err != nil {
		return nil, err
	}
	return &svmGRPCClient{
		client: pb.NewSVMServiceClient(conn),
	}, nil
}

func linearSVCResponse(svmClient *svmGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "linear svc"

		var data pb.LinearRequest
		if err := readRequestData(c, &data); err != nil {
			handleBadRequestError(c, err, http.StatusBadRequest, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := svmClient.client.LinearSVCEvent(ctx, &data)
		if err != nil {
			handleGRPCError(c, err, event_name)
			return
		}

		sendResponse(c, response, event_name)
	}
}

func linearSVRResponse(svmClient *svmGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "linear svr"

		var data pb.LinearRequest
		if err := readRequestData(c, &data); err != nil {
			handleBadRequestError(c, err, http.StatusBadRequest, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := svmClient.client.LinearSVREvent(ctx, &data)
		if err != nil {
			handleGRPCError(c, err, event_name)
			return
		}

		sendResponse(c, response, event_name)
	}
}

func svcResponse(svmClient *svmGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "svc"

		var data pb.SVCRequest
		if err := readRequestData(c, &data); err != nil {
			handleBadRequestError(c, err, http.StatusBadRequest, event_name)
			return
		}

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := svmClient.client.SVCEvent(ctx, &data)
		if err != nil {
			handleGRPCError(c, err, event_name)
			return
		}

		sendResponse(c, response, event_name)
	}
}

File: Javascript/quotes/module/svmService.js
const PROTO_PATH = __dirname + '../lib/AIProto/svm_expression.proto';
const grpc = require('@grpc/grpc-js');
const protoLoader = require('@grpc/proto-loader');
const { defination } = require('./packageDefinitionpackageDefinition')
const { sendStubRequest, handleResponse, handleGRPCError } = require('./helper')

const packageDefinition = protoLoader.loadSync(
    PROTO_PATH,
    defination
);

const svm_proto = grpc.loadPackageDefinition(packageDefinition).svm_expression;

export const createSVMStub = (addr) => {
    return new svm_proto.SVMService(
        addr, 
        grpc.credentials.createInsecure()
    );
};

export const linearSVCResponse = (req, res, stub) => {

    const event_name = `linear svc`

    const request_message = req.body; 

    if (request_message) {
        sendStubRequest(stub, "LinearSVCEvent", request_message)
        .then((response) => handleResponse(response, res))
        .catch((err) => handleGRPCError(err, res, event_name));
    } 
    else {
        handleBadRequestError(res, event_name);
    }

};

export const linearSVRResponse = (req, res, stub) => {

    const event_name = `linear svr`

    const request_message = req.body; 

    if (request_message) {
        sendStubRequest(stub, `LinearSVREvent`, request_message)
        .then((response) => handleResponse(response, res))
        .catch((err) => handleGRPCError(err, res, event_name));
    }
    else {
        handleBadRequestError(res, event_name);
    }
};


export const svcResponse = (req, res, stub) => {

    const event_name = `svc`

    const request_message = req.body;

    if (request_message) {
        sendStubRequest(stub, `SVCEvent`, request_message)
        .then((response) => handleResponse(response, res))
        .catch((err) => handleGRPCError(err, res, event_name));
    }
    else {
        handleBadRequestError(res, event_name);
    }
};
